service/collector/app_resource.go

```diff
@@ -79,24 +113,38 @@ func (c *AppResource) collectAppStatus(ctx context.Context, ch chan<- prometheus
 			app.Status.Version,
 		)
 
-	}
+		if !key.IsCordoned(app) {
+			continue
+		}
 
-}
+		t, err := convertToTime(key.CordonUntil(app))
+		if err != nil {
+			c.logger.Log("level", "warning", "message", "could not convert cordon-until", "stack", fmt.Sprintf("%#v", err))
+			continue
+		}
 
-// Collect is the main metrics collection function.
-func (c *AppResource) Collect(ch chan<- prometheus.Metric) error {
-	ctx := context.Background()
+		ch <- prometheus.MustNewConstMetric(
+			appCordonExpireTimeDesc,
+			prometheus.GaugeValue,
+			float64(t.Unix()),
+			key.AppName(app),
+		)
+	}
+	return nil
+}
 
-	c.logger.LogCtx(ctx, "level", "debug", "message", "collecting metrics")
+func convertToTime(datetime string) (time.Time, error) {
+	layout := "2006-01-02T15:04:05"
 
-	c.collectAppStatus(ctx, ch)
+	split := strings.Split(datetime, ".")
```

Comment:
Sorry more questions. Why is `.` the separator here?

Reply:
This is also from cert-operator that checking the timestamps in secret. AFAIK this is to split millisecond parts from `datetime`. Most of cases we get the timestamps as `2019-12-31T23:59:59.000` but we don't need that part of milliseconds.

Reply:
OK cool that makes sense. For the `cordon-until` we don't care about milliseconds so good to strip them. 👍